### PR TITLE
Docs: document option alias in the options doc

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Noticed that the option documentation missed defining aliases. Moved where sub option definition is since it's a longer section and not always necessary.

## How was this tested?
Read it.

### CODE REVIEW

File: docs/options api.md [hunk 1]
```diff
@@ -13,14 +13,20 @@ need to create:
...
-By style and convention, the internal names should be snake_case. If the option supports having multiple sub_options
-such as Choice options, these can be defined with `option_my_sub_option`, where the preceding `option_` is required and
-stripped for users, so will show as `my_sub_option` in yaml files and if `auto_display_name` is True `My Sub Option`
-on the webhost. All options support `random` as a generic option. `random` chooses from any of the available
-values for that option, and is reserved by AP. You can set this as your default value but you cannot define your own
-new `option_random`.
+By style and convention, the internal names should be snake_case.
 
 ### Option Creation
+- If the option supports having multiple sub_options, such as Choice options, these can be defined with
```
Comment (black-sliver) [inline docs/options api.md:19]: Isn't something like "distinct values" a better term than "sub options"? Not a native speaker, but as programmer sub-option sounds like a dict in a dict to me (i.e. `{"option": {"sub1": "val1", "sub2": "val2"}`).

Comment (alwaysintreble) [reply docs/options api.md:19]: well in code we call them `options` and it creates a dict of them🤔

Comment (black-sliver) [reply docs/options api.md:19]: it's only a dict if the player puts weights on it. the final .value is a distinct value. but yeah the word option is a bit ambiguous in general - my real problem is that `alias_my_sub_option_alt` is just very hard to follow, so i was hoping we might find a better word for it.

**Review by black-sliver — APPROVED** [review]
Thanks, hopefully this is fine. Sadly we don't get a lot of feedback on documentation :S
